staging/src/k8s.io/kubectl/pkg/cmd/logs/logs_nodes.go

```diff
@@ -0,0 +1,584 @@
+}
+
+func (o *NodeLogsOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	if !cmd.Flags().Lookup("unify").Changed {
+		o.Unify = o.Path == "journal"
+	}
+
+	o.Resources = args
+	o.BootChanged = cmd.Flag("boot").Changed
+	o.Raw = o.Raw || o.Path == "journal"
+
+	var err error
+	o.Options, err = o.ToNodeLogOptions()
+	if err != nil {
+		return err
+	}
+
+	o.RESTClientGetter = f
+	o.LogsForObject = polymorphichelpers.LogsForObjectFn
+
+	if o.Object == nil {
+		builder := f.NewBuilder().
+			WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
+			SingleResourceType()
+		if len(o.Resources) > 0 {
+			builder.ResourceNames("nodes", o.Resources...)
+		}
+		if len(o.Role) > 0 {
+			req, err := labels.NewRequirement(fmt.Sprintf("node-role.kubernetes.io/%s", o.Role), selection.Exists, nil)
+			if err != nil {
+				return fmt.Errorf("invalid --role: %v", err)
+			}
+			o.Selector = req.String()
+		}
+		if len(o.Selector) > 0 {
+			builder.ResourceTypes("nodes").LabelSelectorParam(o.Selector)
+		}
+		result := builder.Do()
+
+		o.Prefix = o.Prefix || !result.TargetsSingleItems()
+
+		infos, err := result.Infos()
+		if err != nil {
+			return err
+		}
+
+		if o.Selector == "" && len(infos) != 1 {
+			return fmt.Errorf("expected a node resource")
+		}
+
+		o.Object = infos[0].Object
+		if o.Selector != "" && len(o.Object.(*corev1.NodeList).Items) == 0 {
+			fmt.Fprintln(o.ErrOut, "No node resources found.")
+		}
+	}
+
+	o.ConsumeRequestFn = o.NodeLogsConsumeRequest
+
+	return nil
+}
+
+func (o NodeLogsOptions) Validate() error {
+	logsOptions, ok := o.Options.(*nodev1alpha1.NodeLogOptions)
+	if !ok {
+		return fmt.Errorf("unexpected logs options object")
+	}
+
+	if len(o.Resources) == 0 && len(o.Selector) == 0 {
+		return fmt.Errorf("at least one node name or a selector (-l) must be specified")
+	}
+	if len(o.Resources) > 0 && len(o.Selector) > 0 {
+		return fmt.Errorf("node names and selector may not both be specified")
+	}
+	if logsOptions.TailLines != nil && *logsOptions.TailLines < -1 {
+		return fmt.Errorf("--tail must be greater than or equal to -1")
+	}
+	if o.BootChanged && (o.Boot < -100 || o.Boot > 0) {
+		return fmt.Errorf("--boot accepts values [-100, 0]")
+	}
+
+	return nil
+}
+
+// RunLogs retrieves node logs
+func (o NodeLogsOptions) RunLogs() error {
+	requests, err := o.LogsForObject(o.RESTClientGetter, o.Object, o.Options, o.GetNodeTimeout, false)
+	if err != nil {
+		return err
+	}
+
+	// buffer output for slightly better streaming performance
+	out := bufio.NewWriterSize(o.Out, logsNodesWriterBufferSize)
+	defer out.Flush()
+
+	if o.Unify {
+		// unified output is each source, interleaved in lexographic order (assumes
+		// the source input is sorted by time)
+		return o.parallelConsumeRequest(requests)
+	}
+
+	return o.sequentialConsumeRequest(requests)
+}
+
+func (o NodeLogsOptions) parallelConsumeRequest(requests map[corev1.ObjectReference]rest.ResponseWrapper) error {
```

Comment:
`parallelConsumeRequest` and `sequentialConsumeRequest` are mostly similar to what we already have in https://github.com/kubernetes/kubernetes/blob/0ade4678a7ba527d22f6baa81034dea423267608/staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go#L344-L388 why not refactoring code and re-use instead of duplicating?

Reply:
Add a todo, so we don't forget about it.

Reply:
done